app/assets/javascripts/hidden_until_select.js.coffee

```diff
@@ -0,0 +1,16 @@
+class HiddenUntilSelect
+  constructor: ($root, $hidden_el) ->
+    @$hidden_el = $hidden_el
+    @$controller = $root.find("##{ $hidden_el.attr('data-hide-until-select') }")
+    @$controller.keyup => @checkHide()
+    @$controller.change => @checkHide()
+    @checkHide()
+
+  checkHide: ->
+    @$hidden_el.toggle(@$controller.val())
+
+$ ->
+  # @todo - better scope
+  $scope = $(document)
+  $scope.find("[data-hide-until-select]").each (idx, el) ->
+    new HiddenUntilSelect($scope, $(el))
```

Comment:
So, after looking at Filter and Aidan's upcoming work, and trying to work out the best way to refactor it all to cover this case, I couldn't see how to do it without adding twice as much code. What Filter does is similar but far enough from identical that it requires a load of extra thought to cover this case.

So I'm going to stick with this solution for the moment, and then we can refactor once we get greater overlap in functionality.